pkg/reconciler/taskrun/taskrun.go

```diff
@@ -155,7 +160,12 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 			}
 		}(c.metrics)
 
-		return merr.ErrorOrNil()
+		tr.Status.SetCondition(&apis.Condition{
+			Type:   v1beta1.ConditionCleanedUp,
+			Status: corev1.ConditionTrue,
+			Reason: "Cleaned Up",
+		})
+		return multierror.Append(merr, c.updateStatusLabelsAndAnnotations(tr, original)).ErrorOrNil()
```

Comment:
I think we only need to invoke `updateStatusLabelsAndAnnotations` once per reconcile cycle, and we already do it in L126. It's fine to do it at the end, so as long as we capture and accumulate all possible errors between L126 and here we could remove the call on L126.